lib/common/asar.js

```diff
@@ -562,13 +557,10 @@
       }
 
       const { encoding } = options;
-      const buffer = Buffer.alloc(info.size);
-      const fd = archive.getFd();
-      if (!(fd >= 0)) throw createError(AsarError.NOT_FOUND, { asarPath, filePath });
 
       logASARAccess(asarPath, filePath, info.offset);
-      fs.readSync(fd, buffer, 0, info.size, info.offset);
-      return (encoding) ? buffer.toString(encoding) : buffer;
+      const buffer = Buffer.from(archive.readSync(info.offset, info.size));
```

Comment:
The `Buffer.from` copies data from the argument, can the method return `Buffer` directly to save the extra copy?

Reply:
It does not copy in this case: https://github.com/nodejs/node/blob/090ad7d4a1c32ed6fcd861ce215b1205afe4146c/lib/buffer.js#L293

Reply:
Thanks for the reference, I confirm the content is not copied in this case.

So Node's document of `Buffer.from` is wrong, but it probably does not matter except for low-level library writers.